### ISSUE TITLE
Oracle - Internal Error with RESULT_CACHE

### ISSUE THREAD

**Issue by netsgnut** [sqlfluff/sqlfluff#7003]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A block with `RESULT_CACHE` will result in an internal error.

Link to Oracle PL/SQL Language Reference regarding `RESULT_CACHE`: https://docs.oracle.com/en/database/oracle/oracle-database/23/lnpls/RESULT_CACHE-clause.html

### Expected Behaviour

The block should not return an error.

### Observed Behaviour

An internal error occurs.

```
==== finding fixable violations ====
WARNING    Unable to lint scripts/packages/test.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!                                  
To hide this warning, add the failing file to .sqlfluffignore
RuntimeError: Grammar refers to the 'RELIES_ON' keyword which was not found in the oracle dialect.

The syntax in the query is not (yet?) supported. Try to narrow down your query to a minimal, reproducible case and raise an issue on GitHub.

Or, even better, see this guide on how to help contribute keyword and/or dialect updates:
https://docs.sqlfluff.com/en/stable/perma/contribute_dialect_keywords.html
```

### How to reproduce

Test file:

```
CREATE OR REPLACE PACKAGE test_package IS

FUNCTION any_function_with_result_cache
RETURN VARCHAR2 RESULT_CACHE;

END test_package;
/
```

### Dialect

oracle

### Version

sqlfluff, version 3.4.1 (on Python 3.13)

### Configuration

```
[sqlfluff]
dialect = oracle
templater = jinja
sql_file_exts = .sql,.pks,.pkb
max_line_length = 0

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
ignore_words_regex = ^p_
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
